moveit_ros/moveit_servo/src/servo_calcs.cpp

```diff
@@ -182,25 +192,59 @@ void ServoCalcs::start()
   initial_joint_trajectory->points.push_back(point);
   last_sent_command_ = initial_joint_trajectory;
 
-  timer_ = nh_.createTimer(period_, &ServoCalcs::run, this);
-
   current_state_ = planning_scene_monitor_->getStateMonitor()->getCurrentState();
   tf_moveit_to_ee_frame_ = current_state_->getGlobalLinkTransform(parameters_.planning_frame).inverse() *
                            current_state_->getGlobalLinkTransform(parameters_.ee_frame_name);
   tf_moveit_to_robot_cmd_frame_ = current_state_->getGlobalLinkTransform(parameters_.planning_frame).inverse() *
                                   current_state_->getGlobalLinkTransform(parameters_.robot_link_command_frame);
+
+  stop_requested_ = false;
+  thread_ = std::thread([this] { main(); });
 }
 
-void ServoCalcs::run(const ros::TimerEvent& timer_event)
+void ServoCalcs::main()
 {
-  // Log warning when the last loop duration was longer than the period
-  if (timer_event.profile.last_duration.toSec() > period_.toSec())
+  ros::Rate rate(1.0 / parameters_.publish_period);
+
+  while (ros::ok() && !stop_requested_)
```

Comment:
We should be able to resume the main ServoCalcs() loop easily

Reply:
Also, doesn't `main()` have kind of a special meaning in C++? Not sure this function should be named so

Reply:
main doesn't have a special meaning when it is part of a class... it is a pattern i've seen to use main as the function you start a thread with.

If this function returns it can be joined (the thread is finished and could be joined) which will happen if someone calls start again.